File: Tcm.Domain/Model/School.cs
using System.Collections.Generic;
using Framework.Persistence.Ef;

namespace Tcm.Domain.Model
{
    public class School : EntityBase<int>
    {
        public string Name { get; set; }
        public int CityId { get; set; }
        public City City { get; set; }
        public string Region { get; set; }

        public short SchoolTypeId { get; set; }
        public SchoolType SchoolType { get; set; }

        public short SchoolSubTypeId { get; set; }
        public SchoolSubType SchoolSubType { get; set; }

        public List<SchoolEducationSubCourse> SchoolEducationSubCourses { get; set; }

        /// <summary>
        /// 0:Men , 1:Women
        /// </summary>
        public bool Sex { get; set; }
        /// <summary>
        /// 0:Sobhi , 1:Asri
        /// </summary>
        public short ShiftType { get; set; }

        public int TotalStudentCount { get; set; }
        public int RegisterStudentCount { get; set; }
        public string CreationDate { get; set; }
        public string SchoolNumber { get; set; }
        public string FounderName { get; set; }
        public string ManagerName { get; set; }
        public string PostalAddress { get; set; }
        public string PostalCode { get; set; }
        public string WebUrl { get; set; }

        public string PhoneNumber1 { get; set; }
        public string PhoneNumber2 { get; set; }


    }
    
}

File: Tcm.Interface.WebUi/src/app/pages/school/models/school.ts
import { SchoolEducationSubCourse } from "./SchoolEducationSubCourse";

export class School {
    Id: number;
    Name: string;
    EducationLevelId: number;
    EducationLevelName: string;
    EducationCourseId: number;
    EducationCourseName: string;
    EducationSubCourseName: string;
    ProvinceId: number;
    ProvinceName: string;
    CityId: number;
    CityName: string;
    RegionId: number;
    RegionName: string;
    SchoolTypeId: number;
    SchoolTypeName: string;
    SchoolSubTypeId: number;
    SchoolSubTypeName: string;
    Sex: boolean;
    ShiftType: number;
    TotalStudentCount: string;
    RegisterStudentCount: string;
    SchoolNumber: string;
    FounderName: string;
    ManagerName: string;
    PostalAddress: string;
    PostalCode: string;
    WebUrl: string;
    PhoneNumber1: string;
    PhoneNumber2: string;
    CreationDate: string;
    SchoolEducationSubCourses: SchoolEducationSubCourse[];

    constructor() {
        this.Id = 0; this.Name= ''; this.EducationLevelId= 0; this.EducationLevelName= ''; this.EducationCourseId= 0; this.EducationCourseName= '',
        this.EducationSubCourseName= ''; this.ProvinceId= 0; this.ProvinceName= ''; this.CityId= 0; this.CityName= '',
        this.RegionId= 0; this.RegionName= ''; this.SchoolTypeId= 0; this.SchoolTypeName= ''; this.SchoolSubTypeId= 0; this.CreationDate= '',
        this.SchoolSubTypeName= ''; this.Sex= false; this.ShiftType= 0; this.TotalStudentCount= ''; this.RegisterStudentCount= ''; this.SchoolNumber= '',
        this.FounderName= ''; this.ManagerName= ''; this.PostalAddress= ''; this.PostalCode= ''; this.WebUrl= ''; this.PhoneNumber1= ''; this.PhoneNumber2= ''; 
        this.SchoolEducationSubCourses = null
        
    }
}
